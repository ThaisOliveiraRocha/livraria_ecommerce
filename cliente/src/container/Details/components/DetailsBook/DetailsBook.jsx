import React from "react";
import PropTypes from "prop-types";
import {
  Body,
  ImageComponent,
  Image,
  DetailsComponent,
  Row,
  Col,
  ButtonComponent
} from "./DetailsBook.styles";
import { connect } from "react-redux";
import Button from "../../../../components/Button";
import Titulo from "../../../../components/Title";
import Texto from "../../../../components/Text";
import { addEditCarrinho } from "../../../../store/actions";
import { withRouter } from "react-router-dom";
import Divider from "@material-ui/core/Divider";

export const DetailsBook = ({ livro, addItemCarrinho }) => (
  <Body>
    <ImageComponent>
      <Image src={livro.imagem} />
    </ImageComponent>
    <Divider
      style={{
        width: "1px",
        height: "200px",
        marginLeft: "10px",
        marginRight: "10px",
        backgroundColor: "rgba(0, 0, 0, 0.12)"
      }}
      orientation="vertical"
    />
    <DetailsComponent>
      <Row>
        <Col>
          <Titulo>Título</Titulo>
        </Col>
        <Col>
          <Texto>{livro.titulo}</Texto>
        </Col>
      </Row>
      <Divider
        style={{
          width: "100%",
          height: "1px",

          marginBottom: "5px",
          backgroundColor: "rgba(0, 0, 0, 0.12)"
        }}
        orientation="horizontal"
      />
      <Row>
        <Col>
          <Titulo>Autor</Titulo>
        </Col>
        <Col>
          <Texto>{livro.autor}</Texto>
        </Col>
      </Row>
      <Divider
        style={{
          width: "100%",
          height: "1px",

          marginBottom: "5px",
          backgroundColor: "rgba(0, 0, 0, 0.12)"
        }}
        orientation="horizontal"
      />

      <Row>
        <Col>
          <Titulo>Ano</Titulo>
        </Col>
        <Col>
          <Texto>{livro.ano}</Texto>
        </Col>
      </Row>
      <Divider
        style={{
          width: "100%",
          height: "1px",

          marginBottom: "5px",
          backgroundColor: "rgba(0, 0, 0, 0.12)"
        }}
        orientation="horizontal"
      />

      <Row>
        <Col>
          <Titulo>Situação</Titulo>
        </Col>
        <Col>
          <Texto>{livro.situacao}</Texto>
        </Col>
      </Row>
      <Divider
        style={{
          width: "100%",
          height: "1px",

          marginBottom: "5px",
          backgroundColor: "rgba(0, 0, 0, 0.12)"
        }}
        orientation="horizontal"
      />

      <Row>
        <Col>
          <Titulo>Preço</Titulo>
        </Col>
        <Col>
          <Texto>{livro.preco}</Texto>
        </Col>
      </Row>

      <ButtonComponent>
        <Button onClick={() => addItemCarrinho(livro)}>
          <Texto color="white">Comprar</Texto>
        </Button>
      </ButtonComponent>
    </DetailsComponent>
  </Body>
);

DetailsBook.propTypes = {
  // bla: PropTypes.string,
};

const mapDispatchToProps = dispatch => ({
  addItemCarrinho: livro => dispatch(addEditCarrinho(livro))
});

export default withRouter(connect(null, mapDispatchToProps)(DetailsBook));
