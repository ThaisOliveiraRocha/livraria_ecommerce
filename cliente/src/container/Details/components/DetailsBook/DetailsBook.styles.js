import styled from "styled-components";

export const Body = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 20px;
  height: 250px;
  width: 40%;
  background-color: white;
  border-radius: 5px;
  border: 1px solid #79bac1;
`;

export const ImageComponent = styled.div`
  display: flex;
  align-items: center;
`;

export const DetailsComponent = styled.div`
  display: flex;
  flex-direction: column;
  width: 300px;
  height: 100%;
`;

export const Image = styled.img`
  height: 200px;
  width: auto;
`;

export const Row = styled.div`
  width: 100%;
  height: 30px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
`;

export const Col = styled.div`
  display: flex;
  width: 50%;
  align-items: center;
  justify-content: flex-start;
`;

export const ButtonComponent = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
`;