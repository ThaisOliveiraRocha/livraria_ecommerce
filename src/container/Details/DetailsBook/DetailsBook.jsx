import React from 'react';
import PropTypes from 'prop-types';
//import { Test } from './DetailsBook.styles';

const DetailsBook = (props) => (
  <Layout>
    sdfasdf
  </Layout>
);

DetailsBook.propTypes = {
  // bla: PropTypes.string,
};

DetailsBook.defaultProps = {
  // bla: 'test',
};

export default DetailsBook;
