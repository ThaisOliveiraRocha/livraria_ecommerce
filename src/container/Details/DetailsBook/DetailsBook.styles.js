import styled from "styled-components";

export const Body = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

export const Topo = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const ImageComponent = styled.div`
  display: flex;
`;

export const Image = styled.img`
  width: 200px;
`;


export const InfoComponent = styled.div`
  display: flex;
  flex-direction: column;
`;

export const ButtonCarrinho = styled.div`
  display: flex;
`;