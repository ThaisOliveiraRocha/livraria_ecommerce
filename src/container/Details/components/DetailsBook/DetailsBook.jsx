import React from "react";
import PropTypes from "prop-types";
import {
  Body,
  Topo,
  FichaTecnica,
  DetailsComponent,
  ImageComponent,
  Image,
  InfoComponent,
  LinhaInfo,
  LinhaButton,
  Sinopse,
  AvaliacoesComponent,
  Avaliacao,
  TopoAvaliacao,
  BotoesCabecalho,
  ButtonLike,
  TextoAvaliacaoComponent,
  LivroComponent,
  SinopseTitulo,
  SinopseCorpo
} from "./DetailsBook.styles";
import { connect } from "react-redux";
import Button from "../../../../components/Button";
import Titulo from "../../../../components/Title";
import Text from "../../../../components/Text";
import { FaStar } from "react-icons/fa";
import { addEditCarrinho } from "../../../../store/actions";
import { withRouter } from "react-router-dom";

export const DetailsBook = ({ livro, addItemCarrinho }) => (
  <Body>
    <Topo>
      <FichaTecnica>
        <LivroComponent>
          <ImageComponent>
            <Image src={require(`../../../../assets/images/${livro.imagem}`)} alt=""/>
          </ImageComponent>
          <DetailsComponent>
            <InfoComponent>
              <LinhaInfo>
                <Titulo>Título</Titulo>
              </LinhaInfo>
              <LinhaInfo>
                <Titulo>Autor</Titulo>
              </LinhaInfo>
              <LinhaInfo>
                <Titulo>Ano</Titulo>
              </LinhaInfo>
              <LinhaInfo>
                <Titulo>Situação</Titulo>
              </LinhaInfo>
              <LinhaInfo>
                <Titulo>Preço</Titulo>
              </LinhaInfo>
            </InfoComponent>
            <InfoComponent>
              <LinhaInfo>
                <Text>{livro.titulo}</Text>
              </LinhaInfo>
              <LinhaInfo>
                <Text>{livro.autor}</Text>
              </LinhaInfo>
              <LinhaInfo>
                <Text>{livro.ano}</Text>
              </LinhaInfo>
              <LinhaInfo>
                <Text>{livro.situacao}</Text>
              </LinhaInfo>
              <LinhaInfo>
                <Text>{livro.preco}</Text>
              </LinhaInfo>
            </InfoComponent>
          </DetailsComponent>
        </LivroComponent>
        <LinhaButton>
          <Button className="btnAdicionarCarrinho" onClick={() => addItemCarrinho(livro)}>
            Adicionar ao Carrinho
          </Button>
        </LinhaButton>
      </FichaTecnica>
      <Sinopse>
        <SinopseTitulo>
          <Titulo>Sinopse</Titulo>
        </SinopseTitulo>
        <SinopseCorpo>
          <Text>texto sinopse</Text>
        </SinopseCorpo>
      </Sinopse>
    </Topo>
    <AvaliacoesComponent>
      <Titulo>Avaliações</Titulo>
      <Avaliacao>
        <TopoAvaliacao>
          <Text>Nome user</Text>
          <Text>
            <FaStar className="yellow"></FaStar>
            <FaStar className="yellow"></FaStar>
            <FaStar className="yellow"></FaStar>
            <FaStar className="yellow"></FaStar>
            <FaStar className="yellow"></FaStar>
          </Text>
          <Text>data</Text>
          <BotoesCabecalho>
            <ButtonLike src={require("../../../../assets/images/like.svg")} />
            <ButtonLike src={require("../../../../assets/images/finger.svg")} />
          </BotoesCabecalho>
        </TopoAvaliacao>
        <TextoAvaliacaoComponent>
          <Text>Texto avaliação</Text>
        </TextoAvaliacaoComponent>
      </Avaliacao>
      <Button>Adicionar Comentário</Button>
    </AvaliacoesComponent>
  </Body>
);

DetailsBook.propTypes = {
  // bla: PropTypes.string,
};

const mapDispatchToProps = dispatch => ({
  addItemCarrinho: livro => dispatch(addEditCarrinho(livro))
});

export default withRouter(connect(null, mapDispatchToProps)(DetailsBook));
