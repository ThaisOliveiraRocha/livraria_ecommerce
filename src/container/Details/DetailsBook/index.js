import DetailsBook from './DetailsBook';

export default DetailsBook;
