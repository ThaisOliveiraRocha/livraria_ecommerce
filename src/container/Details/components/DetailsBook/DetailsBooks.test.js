import React from "react";
import { shallow } from "enzyme";
import { DetailsBook } from "./DetailsBook";

const livro = {
  imagem: "book1.png",
  titulo: "É Assim Que Acaba",
  genero: "romance",
  autor: "Colleen Hoover",
  ano: "2018",
  situacao: "novo",
  preco: "40.00",
  qtd: 0
};

describe("<DetailsBook />", () => {
  it("should render without crashing", () => {
    shallow(<DetailsBook livro={livro}/>);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<DetailsBook livro={livro}/>);
    expect(wrapper).toMatchSnapshot();
  });
  it("clicou no botão e chamou a função", () => {
    const mock = jest.fn();
    const wrapper = shallow(
      <DetailsBook livro={livro} addItemCarrinho={mock} />
    );

    wrapper.find(".btnAdicionarCarrinho").simulate("click");

    expect(mock).toHaveBeenCalled();
  });
});
