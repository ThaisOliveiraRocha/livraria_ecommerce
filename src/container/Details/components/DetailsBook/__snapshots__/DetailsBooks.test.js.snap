// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsBook /> should render basic infos 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <styled.div>
        <styled.div>
          <styled.img
            alt=""
            src="book1.png"
          />
        </styled.div>
        <styled.div>
          <styled.div>
            <styled.div>
              <Title>
                Título
              </Title>
            </styled.div>
            <styled.div>
              <Title>
                Autor
              </Title>
            </styled.div>
            <styled.div>
              <Title>
                Ano
              </Title>
            </styled.div>
            <styled.div>
              <Title>
                Situação
              </Title>
            </styled.div>
            <styled.div>
              <Title>
                Preço
              </Title>
            </styled.div>
          </styled.div>
          <styled.div>
            <styled.div>
              <Text>
                É Assim Que Acaba
              </Text>
            </styled.div>
            <styled.div>
              <Text>
                Colleen Hoover
              </Text>
            </styled.div>
            <styled.div>
              <Text>
                2018
              </Text>
            </styled.div>
            <styled.div>
              <Text>
                novo
              </Text>
            </styled.div>
            <styled.div>
              <Text>
                40.00
              </Text>
            </styled.div>
          </styled.div>
        </styled.div>
      </styled.div>
      <styled.div>
        <Button
          className="btnAdicionarCarrinho"
          onClick={[Function]}
        >
          Adicionar ao Carrinho
        </Button>
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <Title>
          Sinopse
        </Title>
      </styled.div>
      <styled.div>
        <Text>
          texto sinopse
        </Text>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    <Title>
      Avaliações
    </Title>
    <styled.div>
      <styled.div>
        <Text>
          Nome user
        </Text>
        <Text>
          <FaStar
            className="yellow"
          />
          <FaStar
            className="yellow"
          />
          <FaStar
            className="yellow"
          />
          <FaStar
            className="yellow"
          />
          <FaStar
            className="yellow"
          />
        </Text>
        <Text>
          data
        </Text>
        <styled.span>
          <styled.img
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "like.svg",
              }
            }
          />
          <styled.img
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "finger.svg",
              }
            }
          />
        </styled.span>
      </styled.div>
      <styled.div>
        <Text>
          Texto avaliação
        </Text>
      </styled.div>
    </styled.div>
    <Button>
      Adicionar Comentário
    </Button>
  </styled.div>
</styled.div>
`;
